### PR TITLE
Add CVE-2017-12617 Apache Tomcat RCE Detector

### PR DESCRIPTION
**Pull request by LeonardoE95**
Hi there,

This PR contains the implementation for the  CVE-2017-12617 detector.
It is worth to note that this detector depends on a new JSP payload configuration. 

Below it is possible to find all the required information for performing the review.

- Issue: 
https://github.com/google/tsunami-security-scanner-plugins/issues/559
- PR Testbed: 
https://github.com/google/security-testbeds/pull/104
- PR Tsunami Scanner: 
https://github.com/google/tsunami-security-scanner/pull/127

### CODE REVIEW

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          + " web.xml configuration, allow unauthenticated actors to upload arbitrary JSP files to"
+          + " the server via a specially crafted request. The uploaded JSP file could then be"
+          + " requested and any code it contained would be executed by the server, leading to"
+          + " Remote Code Execution (RCE).";
+
+  @VisibleForTesting
+  static final String RECOMMENDATION =
+      "Update vulnerable instances and ensure that readonly is set to true for the default servlet,"
+          + " and for the webdav servlet. Block HTTP Methods that allow untrusted users to modify"
+          + " server's resources";
```
Comment (maoning) [inline community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java:82]: Please add upgrade to non-vulnerable versions of Apache Tomcat (versions.....) as part of recommendations as well.

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  @VisibleForTesting
+  static final String VULNERABILITY_REPORT_TITLE = "Apache Tomcat RCE CVE-2017-12617";
+
+  @VisibleForTesting
+  static final String VULN_DESCRIPTION =
+      "Tomcat instances containing a servlet context configured with readonly=false within the"
+          + " web.xml configuration, allow unauthenticated actors to upload arbitrary JSP files to"
+          + " the server via a specially crafted request. The uploaded JSP file could then be"
+          + " requested and any code it contained would be executed by the server, leading to"
+          + " Remote Code Execution (RCE).";
```
Comment (maoning) [inline community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java:76]: Mention the affected apache tomcat versions as well.

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      isVulnerable = response.status().isSuccess() && payload.checkIfExecuted(body);
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log(
+          "Request to target %s failed: could not retrieve JSP file", networkService);
+      return false;
+    }
+
+    // cleanup
+    targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + JSP_FILENAME + "/";
+    logger.atInfo().log("Cleaning JSP file at '%s'", targetUri);
```
Comment (maoning) [inline community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java:165]: "Clean up" or "Remove"

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java (hunk 1 above)
Comment (LeonardoE95) [reply community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java]: Added.

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java (hunk 2 above)
Comment (LeonardoE95) [reply community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java]: Added.

File: community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java (hunk 3 above)
Comment (LeonardoE95) [reply community/detectors/rce/apache_tomcat_cve_2017_12617/src/main/java/com/google/tsunami/plugins/detectors/rce/cve201712617/Cve201712617Detector.java:166]: Fixed.

**Review by maoning — APPROVED** [review]

**Pull request merged**